snuba/web/delete_query.py

```diff
@@ -27,23 +32,62 @@ def construct_condition(body: Dict[str, Any]) -> Expression:
     return combine_and_conditions(and_conditions)
 
 
+def _get_rows_to_delete(
+    storage_key: StorageKey, select_query_to_count_rows: Query
+) -> int:
+    formatted_select_query_to_count_rows = format_query(select_query_to_count_rows)
+    select_query_results = (
+        get_storage(storage_key)
+        .get_cluster()
+        .get_reader()
+        .execute(formatted_select_query_to_count_rows)
+    )
+    return typing.cast(int, select_query_results["data"][0]["count"])
+
+
+def _enforce_max_rows(delete_query: Query) -> None:
+    select_query_to_count_rows = Query(
+        selected_columns=[
+            SelectedExpression("count", FunctionCall("count", "count", ())),
+        ],
+        from_clause=delete_query.get_from_clause(),
+        condition=delete_query.get_condition(),
+    )
+    storage_key = delete_query.get_from_clause().storage_key
+    rows_to_delete = _get_rows_to_delete(
+        storage_key=storage_key, select_query_to_count_rows=select_query_to_count_rows
+    )
+    max_rows_allowed = (
+        get_storage(storage_key).get_deletion_settings().max_rows_to_delete
+    )
+    if rows_to_delete > max_rows_allowed:
+        raise MaxRowsEnforcerException(
+            f"Query wants to delete {rows_to_delete} rows, but the maximum allowed is {max_rows_allowed}"
```

Comment:
I would just want this a little more explicit and wee shorter, something like

`Too many rows to delete (3000), maximum allowed is 1000`